app/frontend/src/store/modules/ches.js

```diff
@@ -53,10 +55,21 @@ export default {
       state.txIds.push(txId);
     },
 
-    ADD_TABLEDATA(state, data) {
+    UPDATE_TABLEDATA(state, data) {
+      // if message already in table..
+      const index = state.tableData.findIndex((message) => message.msgId === data.msgId);
+      if(index > -1 ) {
+        // remove existing message from table
+        state.tableData.splice(index, 1);
+      }
+      // add message to table
       state.tableData.push(data);
     },
 
+    UPDATE_TABLELOADING(state, loading) {
```

Comment:
Optional: This appears to be a set operation, so this mutator could be called `SET_TABLELOADING` instead.